### PR TITLE
[#5404] Update funding agency app to limit queries to United States

### LINKED ISSUE #5404: Limit funding agencies to USA and explore performance enhancements
When attempting to add/update a funding agency, the typeahead does not prioritize results well.
For example, typing "National Science" will not return NSF. Instead it returns a bunch of international funders.
image

image

Also, the typeahead returns more lethargically than in its original implementation. See if there are changes in the Crossreff API
HS v 2.13.3

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #5404 by limiting query to US funders

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. [Enter positive test case here]

### CODE REVIEW

File: theme/static/js/hs-vue/funding-agency-app.js [hunk 1]
```diff
@@ -15,9 +15,16 @@ let fundingAgenciesApp = new Vue({
...
+    HELP_EMAIL: "help@cuahsi.org",
+    CROSSREF_API_URL: "https://api.crossref.org/funders",
+    LIMIT_US: true, // limit to US funders
     MIN_SEARCH_LEN: 3, // min # of chars before running a query
-    DEBOUNCE_API_MS: 500, // debounce api requests
+    // bug in vue-bootstrap-typeahead, this parameter doesn't seem to have any impact
+    // TODO: vue-bootstrap-typeahead is deprecated.
+    // migrate to vue-boostrap-autocomplete or similar
+    // https://github.com/drikusroor/vue-bootstrap-autocomplete
+    MAX_MATCHES: 50, // max number of matches to show
```
Comment (devincowan) [inline theme/static/js/hs-vue/funding-agency-app.js:26]: this PR resolves querying the API but the vue-bootstrap-typeahead does not display the full set

Comment (devincowan) [reply theme/static/js/hs-vue/funding-agency-app.js:26]: I'm not aware of any way to use esm modules like https://github.com/drikusroor/vue-bootstrap-autocomplete for Vue2 non-build static vue apps like the [funding-agency-app](https://github.com/hydroshare/hydroshare/blob/develop/theme/static/js/hs-vue/funding-agency-app.js). @Maurier that's not possible right?

Comment (Maurier) [reply theme/static/js/hs-vue/funding-agency-app.js:26]: I'm afraid this is not possible. They would have to provide the files (script and CSS) via a CDN.

You could try your luck by cloning their repository and building the project and see which files it generates. If they export a UMD build you might be able to use it as in their demo at https://github.com/alexurquhart/vue-bootstrap-typeahead/blob/master/dist/demo.html.

Comment (devincowan) [reply theme/static/js/hs-vue/funding-agency-app.js:26]: thanks! their bundler does export a UMD but when I import it into the HS template it doesn't seem to register any of the components.

Okay thanks Mauriel, looks like we will have to stick with what we have for now

**Review by Maurier — APPROVED** [review]
Looks good.

**Pull request merged**